### PR TITLE
Add fingerprint support to the WAF

### PR DESCRIPTION
**Pull request by manuel-alvarez-alvarez**
# What Does This Do
Adds support for attacker fingerprinting in the WAF according to the [spec](https://docs.google.com/document/d/1DivOa9XsCggmZVzMI57vyxH2_EBJ0-qqIkRHm_sEvSs/edit#heading=h.mdhf4pje9lhk):

- Include the new RC capabilities
- Update to the last libddwaf version with the derivatives field.

# Motivation
Attacker fingerprinting will benefit many different use cases both on the detection and protection side.

# Additional Notes

# Contributor Checklist

- [ ] Format the title [according the contribution guidelines](https://github.com/DataDog/dd-trace-java/blob/master/CONTRIBUTING.md#title-format)
- [ ] Assign the `type:` and (`comp:` or `inst:`) labels in addition to [any usefull labels](https://github.com/DataDog/dd-trace-java/blob/master/CONTRIBUTING.md#labels)
- [ ] Squash your commits prior merging or merge using GitHub's [Squash and merge](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/about-pull-request-merges#squash-and-merge-your-commits)
- [ ] Don't use `close`, `fix` or any [linking keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) when referencing an issue.  
  Use `solves` instead, and assign the PR [milestone](https://github.com/DataDog/dd-trace-java/milestones) to the issue
- [ ] Update the [public documentation](https://docs.datadoghq.com/tracing/trace_collection/library_config/java/) in case of new configuration flag or behavior

Jira ticket: [APPSEC-54547]

[APPSEC-54547]: https://datadoghq.atlassian.net/browse/APPSEC-54547?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by smola — APPROVED** [review]

File: dd-java-agent/appsec/src/test/groovy/com/datadog/appsec/config/AppSecConfigServiceImplSpecification.groovy [hunk 1]
```diff
@@ -261,7 +264,11 @@ class AppSecConfigServiceImplSpecification extends DDSpecification {
       | CAPABILITY_ASM_CUSTOM_RULES
       | CAPABILITY_ASM_CUSTOM_BLOCKING_RESPONSE
       | CAPABILITY_ASM_TRUSTED_IPS
-      | CAPABILITY_ASM_RASP_SQLI)
+      | CAPABILITY_ASM_RASP_SQLI
+      | CAPABILITY_ENDPOINT_FINGERPRINT
+      // | CAPABILITY_ASM_SESSION_FINGERPRINT
```
Comment (jandro996) [inline dd-java-agent/appsec/src/test/groovy/com/datadog/appsec/config/AppSecConfigServiceImplSpecification.groovy:269]: Leftovers?

File: dd-java-agent/appsec/src/test/groovy/com/datadog/appsec/config/AppSecConfigServiceImplSpecification.groovy [hunk 2]
```diff
@@ -406,7 +413,11 @@ class AppSecConfigServiceImplSpecification extends DDSpecification {
...
```
Comment (jandro996) [inline dd-java-agent/appsec/src/test/groovy/com/datadog/appsec/config/AppSecConfigServiceImplSpecification.groovy:418]: Leftovers?

**Review by jandro996 — APPROVED** [review]
LGTM! check my comments related with Leftovers

File: dd-java-agent/appsec/src/test/groovy/com/datadog/appsec/config/AppSecConfigServiceImplSpecification.groovy (hunk 1 above)
Comment (manuel-alvarez-alvarez) [reply dd-java-agent/appsec/src/test/groovy/com/datadog/appsec/config/AppSecConfigServiceImplSpecification.groovy:269]: No, it's another capability that will be added in the next PR

File: dd-java-agent/appsec/src/test/groovy/com/datadog/appsec/config/AppSecConfigServiceImplSpecification.groovy (hunk 2 above)
Comment (manuel-alvarez-alvarez) [reply dd-java-agent/appsec/src/test/groovy/com/datadog/appsec/config/AppSecConfigServiceImplSpecification.groovy:418]: Same as before.

**Pull request merged**